src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VirtualMethodIndexData.cs

```diff
@@ -6,9 +6,37 @@
 namespace Microsoft.Interop
 {
     /// <summary>
-    /// VirtualMethodIndexAttribute data
+    /// Contains the data related to a VirtualMethodIndexAttribute, without references to Roslyn symbols.
+    /// See <seealso cref="VirtualMethodIndexCompilationData"/> for a type with a reference to the StringMarshallingCustomType
     /// </summary>
-    internal sealed record VirtualMethodIndexData(int Index) : InteropAttributeCompilationData
+    internal sealed record VirtualMethodIndexData(int Index) : InteropAttributeData
```

Comment:
The difference between this and the Compilation version is a single `INamedTypeSymbol` field. Is this pattern what Roslyn recommends? It seems very unfortunate to have two types that are identical except for a field with a special "type". The comments are very helpful, but the subtle nature of how to use these types is a bit less than ideal.

Reply:
There's no well-defined pattern for this scenario. This design is following the current best practice of "Provide a data model object that doesn't reference any symbols or anything that would keep the Compilation object alive" (which is the best guidance we have today).

I'm open to alternative ideas/designs that would make this cleaner.

Reply:
Nope, if that is the pattern people are going with sounds good.